src/providers/application/ApplicationProvider.tsx

```diff
@@ -138,6 +140,41 @@ function reducer(
   }
 }
 
+const debouncedSaveFormAndDuos = debounce(
+  async (
+    forms: Form[],
+    dispatch: Dispatch<ApplicationAction>,
+    applicationId: number,
+    handleErrorResponseAfterAction: (response: Response) => Promise<void>
+  ) => {
+    dispatch({ type: ApplicationActionType.LOADING });
```

Comment:
Moving LOADING outside could block the user from doing anything till the debounce is finished. I think this will worsen the user experience where they can only add stuff once previous call is finished

Reply:
<img width="837" alt="Screenshot 2024-08-22 at 12 24 17" src="https://github.com/user-attachments/assets/3f1ec611-3e51-41ea-bc9a-ae4c9c1bdd92">
<img width="851" alt="Screenshot 2024-08-22 at 12 24 26" src="https://github.com/user-attachments/assets/4f070137-7ee7-4f47-b6e1-3c6394f52307">